pkg/controlplane/control/manager.go

```diff
@@ -41,12 +46,78 @@ type Manager struct {
 	crdMode bool
 	ports   *portManager
 
-	logger *logrus.Entry
+	podLock sync.RWMutex
+	// clDataplanePodInfo stores the IPv4/v6 address of cl-dataplane pod per namespace
+	clDataplanePodInfo map[string]string
+	logger             *logrus.Entry
 }
 
-// AddImport adds a listening socket for an imported remote service.
-func (m *Manager) AddImport(ctx context.Context, imp *v1alpha1.Import) error {
-	m.logger.Infof("Adding import '%s/%s'.", imp.Namespace, imp.Name)
+func (m *Manager) addImportEndpointSlice(ctx context.Context, imp *v1alpha1.Import) error {
+	m.logger.Infof("Adding import endpointslice '%s/%s'.", imp.Namespace, imp.Name)
+
+	protocol := v1.ProtocolTCP
+	var port32 int32
+	newEndpointslice := discv1.EndpointSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      imp.Name + "-" + uuid.NewString()[:5],
+			Namespace: imp.Namespace,
+			Labels: map[string]string{
+				"kubernetes.io/service-name":             imp.Spec.Merge,
+				"endpointslice.kubernetes.io/managed-by": cpapp.Name,
+			},
+		},
+		AddressType: discv1.AddressTypeIPv4,
+		Endpoints: []discv1.Endpoint{
+			{
+				Addresses: []string{m.clDataplanePodInfo[imp.Namespace]},
+			},
+		},
+	}
+	var oldEndpointslice discv1.EndpointSlice
+	var create bool
+	err := m.client.Get(
```

Comment:
Yes you are right. I think I was in two minds of adding multiple imports of same service (different peers) in the same endpointslice, However it would make it complicated to delete. Hence, finally decided to have an endpointslice per import from peer.

Reply:
Not sure I follow why need per peer. It should resolve to clusterlink which takes care of the peer multiplicity

Reply:
Yes, Correct, I have changed the logic so will not be getting and updating endpointslice, since listener remains the same.